modules/ServerAPI/src/client/Thing.coffee

```diff
@@ -187,6 +189,17 @@ class Thing extends Backbone.Model
 			if !data.has ('recordedDate') || data.get('recordedDate') == null
 				data.set recordedDate: rDate
 
+	setName: =>
+		# This is a best guess as a pretty name using the ls type and kind
+		# it can be overriden by setting the name of the ls thing when
+		# instantiating the ls thing
+		if !@name?
+			typeName = @.get("lsType").charAt(0).toUpperCase() + @.get("lsType").slice(1).toLowerCase();
+			kindName = @.get("lsKind").charAt(0).toUpperCase() + @.get("lsKind").slice(1).toLowerCase();
+			@name = "#{kindName} #{typeName}"
```

Comment:
I found that `"#{kindName} #{typeName}"` was typically a better looking name than "#{typeName} #{kindName}" but this can always be overridden by passing the name when instatiating the new thing (or by setting it in the thing conf)